### ISSUE TITLE
[BUG]: Custom metrics return 0.0000

### ISSUE THREAD

**Issue by jshahbazi** [pycaret/pycaret#3210]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

In 3.0.0rc6, on an M1 macOS, whenever I use add_metric, the function only returns 0.0000.

### Reproducible Example

```python
from sklearn.metrics import log_loss
add_metric('logloss', 'Log Loss', log_loss, greater_is_better = False)
```

### Expected Behavior

The function should return a value.

### Actual Results

```python-traceback
All metric results are 0.0000
```

### Installed Versions

<details>

```
System:

System:
    python: 3.10.8 (main, Oct 13 2022, 09:48:40) [Clang 14.0.0 (clang-1400.0.29.102)]
executable: /opt/homebrew/opt/python@3.10/bin/python3.10
   machine: macOS-12.6.1-arm64-arm-64bit

PyCaret required dependencies:
                 pip: 22.3.1
          setuptools: 65.4.1
             pycaret: 3.0.0rc6
             IPython: 8.7.0
          ipywidgets: 8.0.4
                tqdm: 4.64.1
               numpy: 1.22.0
              pandas: 1.3.5
              jinja2: 3.1.2
               scipy: 1.10.0rc1
              joblib: 1.2.0
             sklearn: 1.1.3
                pyod: 1.0.7
            imblearn: 0.10.0
   category_encoders: 2.5.1.post0
            lightgbm: 3.3.3
               numba: 0.56.4
            requests: 2.28.1
          matplotlib: 3.6.2
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.11.0
             kaleido: 0.2.1
         statsmodels: 0.13.5
              sktime: 0.14.1
               tbats: 1.1.2
            pmdarima: 2.0.2
              psutil: 5.9.4

PyCaret optional dependencies:
                shap: Not installed
           interpret: Not installed
                umap: Not installed
    pandas_profiling: Not installed
  explainerdashboard: Not installed
             autoviz: Not installed
           fairlearn: Not installed
             xgboost: 1.7.2
            catboost: Not installed
              kmodes: Not installed
             mlxtend: Not installed
       statsforecast: Not installed
        tune_sklearn: Not installed
                 ray: Not installed
            hyperopt: Not installed
              optuna: Not installed
               skopt: Not installed
              mlflow: Not installed
              gradio: Not installed
             fastapi: Not installed
             uvicorn: Not installed
              m2cgen: Not installed
           evidently: Not installed
                nltk: 3.6.7
            pyLDAvis: Not installed
              gensim: Not installed
               spacy: Not installed
           wordcloud: Not installed
            textblob: Not installed
               fugue: Not installed
           streamlit: Not installed
             prophet: Not installed
```

</details>

**Comment by ngupta23** [comment]
Could you provide the complete reproducible example  (self contained with code and data) so we can debug the issue? Thanks!

**Comment by jshahbazi** [comment]
I'm following along this notebook:  [https://nbviewer.org/github/gofornaman/Predicting-Customer-Churn-using-PyCaret/blob/main/notebook/model_v1.ipynb](https://nbviewer.org/github/gofornaman/Predicting-Customer-Churn-using-PyCaret/blob/main/notebook/model_v1.ipynb)

**Comment by jshahbazi** [comment]
And then when I go on my own and add the logloss metric, this is what comes up:
<img width="1187" alt="image" src="https://user-images.githubusercontent.com/4032258/209980377-5dc5e32b-a577-4b5a-87b3-053302f76f7b.png">

**Comment by ngupta23** [comment]
@Yard1 I am able to reproduce this issue. See notebook: https://gist.github.com/ngupta23/8bd712f65d8f70630967aec6dde4266f

**Comment by Yard1** [comment]
@jshahbazi The issue here is that the metrics function gets untransformed labels as `y` and `y_pred`, meaning those dataframes have "Yes" and "No" instead of 1 and 0. Simply modifying the function to:
```python
def calculate_profit(y, y_pred):
    y = y.replace({"Yes": 1, "No": 0}).astype(int)
    y_pred = y_pred.replace({"Yes": 1, "No": 0}).astype(int)
    tp = np.where((y_pred==1) & (y==1), (5000-1000), 0)
    fp = np.where((y_pred==1) & (y==0), -1000, 0)
    return np.sum([tp,fp])# add metric to PyCaret
```
makes it work as expected. Hope that helps - please reopen the issue if you have any more questions regarding this.
